### PR TITLE
Fix framing rustdocs

### LINKED ISSUE #769: Framing rustdocs

### PR DESCRIPTION
**Pull request by raphaelrobert**
Fixes #769.

Note that I could turn some types into `pub(crate)` and I'm therefore only touching the public docs.

### CODE REVIEW

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
I'm fine with the changes. I just wonder if there should be a little more high-level information for the framing module.

image

**Comment by franziskuskiefer** [comment]
Also, there are `messages` and `framing` as top-level modules right now. It's not clear why we need both from the docs.

File: openmls/src/framing/validation.rs [hunk 1]
```diff
@@ -403,15 +405,27 @@ impl VerifiedExternalMessage {
     }
 }
 
-/// Message that contains messages that are syntactically and semantically correct.
-/// [StagedCommit] and [QueuedProposal] can be inspected for authorization purposes.
+/// Message that has passed all syntax and semantics checks.
+///
+/// See the vriants' documentation for mor information.
```
Comment (kkohbrok) [inline openmls/src/framing/validation.rs:410]:
```suggestion
/// See the variants' documentation for more information.
```

File: openmls/src/framing/validation.rs [hunk 2]
```diff
@@ -403,15 +405,27 @@ impl VerifiedExternalMessage {
     }
 }
 
-/// Message that contains messages that are syntactically and semantically correct.
-/// [StagedCommit] and [QueuedProposal] can be inspected for authorization purposes.
+/// Message that has passed all syntax and semantics checks.
```
Comment (kkohbrok) [inline openmls/src/framing/validation.rs:408]:
```suggestion
/// A message that has passed all syntax and semantics checks.
```

File: openmls/src/framing/validation.rs [hunk 3]
```diff
@@ -403,15 +405,27 @@ impl VerifiedExternalMessage {
...
+    ///
+    /// The [`QueuedProposal`] can be inspected for authorization purposes by the application.
+    /// If the proposal is deemed to be allowed, it should be added to the group's proposal
+    /// queue using [`MlsGroup::store_pending_proposal()`].
     ProposalMessage(Box<QueuedProposal>),
-    /// Staged Commit
+    /// A Commit message.
+    ///
+    /// The [`StagedCommit`] can be inspected for authorization purposes by the application.
+    /// If the type of the commit and the proposals it covers aredeemed to be allowed,
```
Comment (kkohbrok) [inline openmls/src/framing/validation.rs:427]:
```suggestion
    /// If the type of the commit and the proposals it covers are deemed to be allowed,
```

**Review by kkohbrok — APPROVED** [review]
Just a few typos, otherwise fine by me.

**Review by franziskuskiefer — APPROVED** [review]

**Pull request merged**